training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -44,8 +43,22 @@ def __init__(self, model_id: str):
         """
         self._model_id = model_id
         self._model_uri = self.HF_ENDPOINT + f"/{model_id}"
-        self._model_info = None
         self._hf_api = HfApi(endpoint=self.HF_ENDPOINT)
+        self._is_valid_id = None
+        self._model_info = None
+
+    @property
+    def is_valid_id(self):
+        """Returns true if Hugging face model id is valid."""
+        if not self._is_valid_id:
```

Comment:
You probably don't want this to run again after `self._is_valid_id` is set to `False`:
```suggestion
        if self._is_valid_id is None:
```